File: apps/synapse-web/src/domain/entities/inventory.ts
export interface StockLevel {
	material_name: string;
	warehouse: string;
	quantity: number;
	unit: string;
	reorder_point: number;
	is_low: boolean;
	last_updated: string;
}

export interface MaterialMovement {
	id: string;
	material_name: string;
	movement_type: "IN" | "OUT" | "TRANSFER";
	quantity: number;
	warehouse: string;
	date: string;
	reference: string;
}

export interface StockLevelListResponse {
	items: StockLevel[];
	total: number;
}

export interface MovementListResponse {
	items: MaterialMovement[];
	total: number;
}

File: apps/synapse-api/synapse_api/presentation/schemas/inventory.py
from pydantic import BaseModel


class StockLevelSchema(BaseModel):
    material_name: str
    warehouse: str
    quantity: int
    unit: str
    reorder_point: int
    is_low: bool
    last_updated: str


class MaterialMovementSchema(BaseModel):
    id: str
    material_name: str
    movement_type: str
    quantity: int
    warehouse: str
    date: str
    reference: str


class StockLevelListResponse(BaseModel):
    items: list[StockLevelSchema]
    total: int


class MovementListResponse(BaseModel):
    items: list[MaterialMovementSchema]
    total: int
